src/foundry/client-esm/dice/terms/dice.d.mts

```diff
@@ -0,0 +1,398 @@
+   * @param command - The parsed modifier command
+   * @param modifier -  The full modifier request
+   */
+  protected _evaluateModifier(command: string, modifier: string): void;
+  /* -------------------------------------------- */
+
+  /**
+   * A helper comparison function.
+   * Returns a boolean depending on whether the result compares favorably against the target.
+   * @param result     - The result being compared
+   * @param comparison - The comparison operator in [=,&lt;,&lt;=,\>,\>=]
+   * @param target     - The target value
+   * @returns Is the comparison true?
+   */
+  static compareResult(result: number, comparison: string, target: number): boolean;
+
+  /* -------------------------------------------- */
+
+  /**
+   * A helper method to modify the results array of a dice term by flagging certain results are kept or dropped.
+   * @param results - The results array
+   * @param number  - The number to keep or drop
+   * @param keep    - Keep results?
+   *                  (default: `true`)
+   * @param highest - Keep the highest?
+   *                  (default: `true`)
+   * @returns The modified results array
+   */
+  protected static _keepOrDrop(
+    results: DiceTerm.Result[],
+    number: number,
+    {
+      keep,
+      highest,
+    }?: {
+      keep: boolean;
+      highest: boolean;
+    },
+  ): DiceTerm.Result;
+
+  /* -------------------------------------------- */
+
+  /**
+   * A reusable helper function to handle the identification and deduction of failures
+   * @param flagFailure - (default: `false`)
+   * @param flagSuccess - (default: `false`)
+   */
+  protected static _applyCount(
+    results: DiceTerm.Result[],
+    comparison: string,
+    target: number,
+    {
+      flagSuccess,
+      flagFailure,
+    }?: {
+      flagSuccess: boolean;
+      flagFailure: boolean;
+    },
+  ): void;
+
+  /* -------------------------------------------- */
+
+  /**
+   * A reusable helper function to handle the identification and deduction of failures
+   * @param deductFailure - (default: `false`)
+   * @param invertFailure - (default: `false`)
+   */
+  protected static _applyDeduct(
+    results: DiceTerm.Result[],
+    comparison: string,
+    target: number,
+    {
+      deductFailure,
+      invertFailure,
+    }?: {
+      deductFailure: boolean;
+      invertFailure: boolean;
+    },
+  ): void;
+
+  /* -------------------------------------------- */
+  /*  Factory Methods                             */
+  /* -------------------------------------------- */
+
+  /**
+   * Determine whether a string expression matches this type of term
+   * @param expression - The expression to parse
+   * @param options - Additional options which customize the match
+   * @param imputeNumber - Allow the number of dice to be optional, i.e. "d6"
+   *                       (default: `true`)
+   */
+  static matchTerm(expression: string, { imputeNumber }?: { imputeNumber: boolean }): RegExpMatchArray | null;
```

Comment:
impute is a surprisingly deep word to pull form instead of `optional` or something lol. No action necessary, this is Foundr'ys doing anyways.

Reply:
xD